clarifai/runners/utils/data_handler.py

```diff
@@ -1,231 +1,248 @@
+            part.data.parts.add().data.image.CopyFrom(part_item.to_proto())
+          elif isinstance(part_item, Audio):
+            part.data.parts.add().data.audio.CopyFrom(part_item.to_proto())
+          elif isinstance(part_item, Video):
+            part.data.parts.add().data.video.CopyFrom(part_item.to_proto())
+          elif isinstance(part_item, str):
+            part.data.parts.add().data.text.raw = part_item
+          elif isinstance(part_item, bytes):
+            part.data.parts.add().data.bytes_value = part_item
+          elif isinstance(part_item, int):
+            part.data.parts.add().data.int_value = part_item
+          elif isinstance(part_item, float):
+            part.data.parts.add().data.float_value = part_item
+          elif isinstance(part_item, bool):
+            part.data.parts.add().data.boolean = part_item
+          elif isinstance(part_item, np.ndarray):
+            ndarray_proto = resources_pb2.NDArray(
+                buffer=part_item.tobytes(), shape=part_item.shape, dtype=str(part_item.dtype))
+            part.data.parts.add().data.ndarray.CopyFrom(ndarray_proto)
+          elif isinstance(part_item, PILImage.Image):
+            image = Image.from_pil(part_item)
+            part.data.parts.add().data.image.CopyFrom(image.to_proto())
+          else:
+            raise TypeError(f"Unsupported Output type {type(part_item)} for part '{part_name}'")
+    elif isinstance(part_value, dict):
+      dict_to_metadata(part.data, part_value)
+    else:
+      raise TypeError(f"Unsupported Output type {type(part_value)} for part '{part_name}'")
+
+  return data_proto
+
+
+def proto_to_kwargs(data: resources_pb2.Data) -> dict:
+  """Converts the Clarifai protobuf Data message to a dictionary."""
+  kwargs = {}
+  for part in data.parts:
+    part_name = part.id
+    part_data = part.data
+    if part_data.HasField("text"):
+      kwargs[part_name] = Text.from_proto(part_data.text)
+    elif part_data.HasField("image"):
+      kwargs[part_name] = Image(part_data.image)
+    elif part_data.HasField("audio"):
+      kwargs[part_name] = Audio(part_data.audio)
+    elif part_data.HasField("video"):
+      kwargs[part_name] = Video(part_data.video)
+    elif part_data.HasField("bytes_value"):
+      kwargs[part_name] = part_data.bytes_value
+    elif part_data.HasField("int_value"):
+      kwargs[part_name] = part_data.int_value
+    elif part_data.HasField("float_value"):
+      kwargs[part_name] = part_data.float_value
+    elif part_data.HasField("boolean"):
+      kwargs[part_name] = part_data.boolean
+    elif part_data.HasField("ndarray"):
+      ndarray = part_data.ndarray
+      kwargs[part_name] = np.frombuffer(
+          ndarray.buffer, dtype=np.dtype(ndarray.dtype)).reshape(ndarray.shape)
+    elif part_data.HasField("metadata"):
+      kwargs[part_name] = metadata_to_dict(part_data)
+    elif part_data.parts:
+      kwargs[part_name] = []
+      for part_item in part_data.parts:
```

Comment:
same thing here, re-use a function so each field doesn't have to be updated in two places.